src/libUtils/UpgradeManager.h

```diff
@@ -17,58 +17,20 @@
 #ifndef __UPGRADEMANAGER_H__
 #define __UPGRADEMANAGER_H__
 
-#include "common/Serializable.h"
+#include "libMediator/Mediator.h"
+#include "libUtils/SWInfo.h"
 #include <cstring>
+#include <curl/curl.h>
 #include <memory>
-#include <stdint.h>
 #include <string>
 
-class SWInfo : public Serializable
-{
-    uint32_t m_major;
-    uint32_t m_minor;
-    uint32_t m_fix;
-    uint64_t m_upgradeDS;
-    uint32_t m_commit;
-
-public:
-    static const unsigned int SIZE = sizeof(uint32_t) + sizeof(uint32_t)
-        + sizeof(uint32_t) + sizeof(uint64_t) + sizeof(uint32_t);
-
-    /// Default constructor for uninitialized version information.
-    SWInfo();
-
-    /// Constructor.
-    SWInfo(const uint32_t& major, const uint32_t& minor, const uint32_t& fix,
-           const uint64_t& upgradeDS, const uint32_t& commit);
-
-    /// Destructor.
-    ~SWInfo();
-
-    /// Copy constructor.
-    SWInfo(const SWInfo&);
-
-    /// Implements the Serialize function inherited from Serializable.
-    unsigned int Serialize(std::vector<unsigned char>& dst,
-                           unsigned int offset) const;
-
-    /// Implements the Deserialize function inherited from Serializable.
-    int Deserialize(const std::vector<unsigned char>& src, unsigned int offset);
-
-    /// Less-than comparison operator.
-    bool operator<(const SWInfo& r) const;
-
-    /// Greater-than comparison operator.
-    bool operator>(const SWInfo& r) const;
-
-    /// Equality operator.
-    bool operator==(const SWInfo& r) const;
-};
-
 class UpgradeManager
 {
 private:
-    std::shared_ptr<SWInfo> m_curSWInfo;
+    std::shared_ptr<SWInfo> m_latestSWInfo;
+    std::vector<unsigned char> m_latestSHA;
+    CURL* m_curl;
```

Comment:
we can also wrap `m_curl` in a `unique_ptr` to make life easier.

Reply:
Maybe need more time to investigate how to use unique_ptr with CURL because it's a "typedef."